compiler/src/main/kotlin/ast/ast.kt

```diff
@@ -206,13 +206,20 @@ class ArrayLookupExpr(
 
 /**
  * @param ctx the context from witch this node was created from
- * @param values Elements of the array.
+ * @param body Elements of the array.
  * @param declaredType Type listed before the body/values of the array.
  */
-class ArrayBodyExpr(
+class SizedArrayExpr(
+    ctx: SourceContext,
+    val body: ArrayLiteralExpr?,
+    val declaredType: Type,
+    var declaredSize: List<Int?>
+) : Expr(ctx)
+
+class ArrayLiteralExpr(
     ctx: SourceContext,
     val values: List<Expr>,
-    val declaredType: Type
+    var size: Int = -1
```

Comment:
Doesn't need a default value.

Reply:
```suggestion
    var size: Int? = null
```
Since the actual size isn't guaranteed to be know prior to type checking, we need to give it some value when it's instantiated during the reduce phase

Reply:
That's true. I prefer `Int?` then